lib/commands/cryptographic-identities.ts

```diff
@@ -602,11 +604,32 @@ export class ImportCryptographicIdentity implements ICommand {
 			}
 
 			if(!password && importType === CryptographicIdentityConstants.PKCS12CERTIFICATE) {
-				password = this.$prompter.getPassword("Certificate file password", { allowEmpty: true }).wait();
+				isPasswordRequired = true;
 			}
 
-			var targetFile = this.$fs.createReadStream(certificateFile);
-			var result = this.$server.identityStore.importIdentity(<any>importType, password, targetFile).wait();
+			var targetFile : any;
+			var result : Server.CryptographicIdentityData[];
+
+			for (var i = 0; i < CryptographicIdentityConstants.MAX_ALLOWED_PASSWORD_ATTEMPTS; ++i) {
+				var noErrorOccurred = true;
+				
+				targetFile = this.$fs.createReadStream(certificateFile);
+				if (isPasswordRequired) {
+					password = this.$prompter.getPassword("Certificate file password", { allowEmpty: true }).wait();
+				}
+
+				try {
+					result = this.$server.identityStore.importIdentity(<any>importType, password, targetFile).wait();
+				}
+				catch(err) {
+					noErrorOccurred = false;
+					this.$logger.error(err.message);
+				}
+
+				if (noErrorOccurred) {
+					break;
+				}
+			}
 
```

Comment:
Just one idea - if the three calls fail, the code continues. Currently this is not a problem, but maybe it is a good idea to call $errors.fail (...) in such case